cpplint.py

```diff
@@ -934,9 +934,93 @@ def unicode_escape_decode(x):
 # This is set by --headers flag.
 _hpp_headers = set([])
 
-# {str, bool}: a map from error categories to booleans which indicate if the
-# category should be suppressed for every line.
-_global_error_suppressions = {}
+class ErrorSuppressions:
+  """Class to track all error suppressions for cpplint"""
+
+  class SuppressionRange:
+    """Class to represent a range of line numbers for which an error is suppressed"""
+    def __init__(self, begin=0, end=math.inf):
+      self._begin = begin
+      self._end = end
+
+    def __str__(self):
+      return f'[{self.begin}-{self.end}]'
+
+    def __contains__(self, obj):
+      if isinstance(obj, ErrorSuppressions.SuppressionRange):
+        return self.begin <= obj.begin and self.end >= obj.end
+      elif isinstance(obj, int):
+        return self.begin <= obj <= self.end
+      raise TypeError()
+
+    @property
```

Comment:
nit: I am not against using decorators in python, but they seem to stick out in cpplint.py because the older code does not use any. So they could be left out for consistencies sake, but I would also accept if you want to use this style.

Reply:
Removed